### PR TITLE
Allow LoadFrom to load assemblies in TPA

### PR DESCRIPTION
**Pull request by rahku**

### CODE REVIEW

File: src/binder/clrprivbindercoreclr.cpp [hunk 1]
```diff
@@ -155,6 +155,7 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 // The simple name of the assembly being requested to be bound was found in the TPA list.
                 // Now, perform the actual bind to see if the assembly was really in the TPA assembly list or not.
                 hr = BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, true /* excludeAppPaths */);
+#ifndef FEATURE_CORECLR
```
Comment (jkotas) [inline src/binder/clrprivbindercoreclr.cpp]: You can just delete this code. This whole file is CoreCLR only.

**Comment by davidfowl** [comment]
What are the semantics of LoadFrom on coreclr?

**Comment by jkotas** [comment]
Just forwards to `AssemblyLoadContext.Default.LoadFromAssemblyPath`.

https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Reflection/Assembly.cs#L118

**Comment by davidfowl** [comment]
Doesn't it need to handle loading dependencies from the same location as the original dll?

**Comment by jkotas** [comment]
It does not replicate desktop LoadFrom context with all its tricky behaviors ... it is hard to do it without bringing verbatim copy of fusion ... .

**Comment by davidfowl** [comment]
@jkotas it needs to at least load dependencies, otherwise this will be a compat nightmare. The API will be there but you'll still need to hook resolve to make the behavior consistent.

**Comment by jkotas** [comment]
LoadFrom has always been a nightmare. There are hundreds of articles and discussions written about people getting burned by it (e.g. https://blogs.msdn.microsoft.com/aszego/2009/10/16/avoid-using-assembly-loadfrom/ to just ) and getting to a conclusion that this API should be avoided.

If we make LoadFrom to automatically look for dependencies in the same directly, it will fix some of the uses and break others. The difference is that folks can add code that looks for the dependencies to workaround the issue. But if we look for the dependencies by default and it does not work, folks cannot undo it.

**Comment by davidfowl** [comment]
It doesn't have to be all or nothing. In my experience, that's the most observable difference between the 2.

**Comment by rahku** [comment]
will open another PR soon

**Comment by davidfowl** [comment]
@raku, was there another PR?

**Pull request closed without merging**